game-core/src/test/java/org/triplea/live/servers/ServerYamlValidityTest.java

```diff
@@ -0,0 +1,98 @@
+import static org.triplea.test.common.TestDataFileReader.readContents;
+
+import feign.template.UriUtils;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Objects;
+import java.util.function.Supplier;
+import org.hamcrest.core.StringEndsWith;
+import org.junit.jupiter.api.Test;
+import org.triplea.util.Version;
+
+/**
+ * Checks the current 'servers.yml' to verify it is semantically valid. eg:
+ *
+ * <ul>
+ *   <li>Contains a latest version, greater than 0.0
+ *   <li>Inactive versions have null URI and no message
+ *   <li>Active versions have a URI and a message
+ *   <li>URI does not end with a trailing slash
+ *   <li>URI is absolute, it contains protocol and host
+ *   <li>No exact version duplicates
+ * </ul>
+ */
+class ServerYamlValidityTest {
+  private static final Supplier<InputStream> SERVERS_YML =
+      () -> asInputStream(readContents("servers.yml"));
+  private static final LiveServers liveServers = new ServerYamlParser().apply(SERVERS_YML.get());
+
+  @Test
+  void verifyLatestVersion() {
+    assertThat(liveServers.getLatestEngineVersion(), notNullValue());
+    assertThat(liveServers.getLatestEngineVersion().isGreaterThan(new Version("0.0")), is(true));
+  }
+
+  @Test
+  void uriValuesDoNotEndInSlash() {
+    liveServers.getServers().stream()
+        .map(ServerProperties::getUri)
+        .filter(Objects::nonNull)
+        .map(URI::toString)
+        .forEach(uriString -> assertThat(uriString, not(StringEndsWith.endsWith("/"))));
+  }
+
+  @Test
+  void uriValuesAreAbsolute() {
+    liveServers.getServers().stream()
+        .map(ServerProperties::getUri)
+        .filter(Objects::nonNull)
+        .map(URI::toString)
+        .forEach(uri -> assertThat(UriUtils.isAbsolute(uri), is(true)));
+  }
+
+  @Test
+  void nonNullUrisAreMarkedAsActive() {
+    liveServers.getServers().stream()
+        .filter(props -> props.getUri() != null)
+        .forEach(props -> assertThat(props.isInactive(), is(false)));
+  }
+
+  @Test
+  void nullUrisAreMarkedAsInactive() {
+    liveServers.getServers().stream()
+        .filter(props -> props.getUri() == null)
+        .forEach(props -> assertThat(props.isInactive(), is(true)));
+  }
+
+  @Test
+  void inactiveVersionsDoNotHaveAMessage() {
+    liveServers.getServers().stream()
+        .filter(ServerProperties::isInactive)
+        .forEach(props -> assertThat(props.getMessage(), emptyString()));
+  }
+
+  @Test
+  void activeVersionsHaveAMessage() {
+    liveServers.getServers().stream()
+        .filter(props -> !props.isInactive())
+        .forEach(props -> assertThat(props.getMessage(), not(emptyString())));
+  }
+
+  @Test
+  void noVersionDuplicates() {
+    final long uniqCount =
+        liveServers.getServers().stream()
+            .map(ServerProperties::getMinEngineVersion)
+            .distinct()
+            .count();
+    assertThat((long) liveServers.getServers().size(), is(uniqCount));
```

Comment:
Ok, didn't expect this cast to long here, but it makes somewhat sense

Reply:
Indeed, same here.

Would be nice to have a `countInt` method, also a bit a weakness of hamcrest that leaving the cast off yields no compilation error, yet the assertion fails due to mismatched type. Might be argument for us to create a numeric matcher that will do a value comparison of long to int.